src/pages/DatasetRegistration.js

```diff
@@ -769,6 +769,7 @@ class DatasetRegistration extends Component {
                         name: 'researcher',
                         id: 'inputResearcher',
                         value: this.state.datasetData.researcher,
+                        onChange: this.handleChange,
```

Comment:
If you're using text inputs and they're not tied to a real-time update feature it's better to use onBlur. This way instead of firing the handleChange function on every keystroke you'll fire it once the text input is out of focus.

Reply:
Great point ... I would have missed this.